### ISSUE TITLE
Some tippers dont tip at tip triggers (A fix for it)

### ISSUE THREAD

**Issue by Satissis** [netjungle/courseplay#55]
Hi.

As you might know, then some tippers wont tip at a tip trigger since fs2013 was released.
Even some tippers froze totally and you had to reload the game to get it to work again.

This was due to you had to disable the:

``` lua
--if not tipper.allowsDetaching then
```

in tippers.lua (function: courseplay:unload_tippers(self)), since that didn't work anymore.

Well I found the solution to this, since I figured out how one of the new functions was working.

Replace the:

``` lua
--if not tipper.allowsDetaching then
```

with:

``` lua
if self.currentTipTrigger:getTipDistanceFromTrailer(tipper, tipper.currentTipReferencePointIndex) == 0 then
```

This will make the trailer wait to try tip until it's actually inside the tip area, where it's allowed to tip.

I have tested this with one of the tippers that froze when using driver helper mode and it works perfectly.

The only remaining thing that needs to be adjusted now is when the tractor + tipper is leaving the tip trigger, it will still drive very slow and go up and down in speed until the tractor have cleared the tip trigger search area.

Kind regards
Satis

**Comment by ThomasGaertner** [comment]
thanks for that hint, maybe I can solve all my "triggerproblems" with it ( also the slow speed)

**Comment by JakobTischler** [comment]
fixed with commit [f8d0eb4f94](https://github.com/netjungle/courseplay/commit/f8d0eb4f940de79a9402d45288517e62bcef9eb8)
